spyder/plugins/editor/widgets/editor.py

```diff
@@ -1812,7 +1812,7 @@ def _write_to_file(self, fileinfo, filename):
         This is a low-level function that only saves the text to file in the
         correct encoding without doing any error handling.
         """
-        txt = fileinfo.editor.get_text_with_eol()
+        txt = to_text_string(fileinfo.editor.get_text_with_eol())

```

Comment:
I don't think this is correct because `to_text_string` converts strings to unicode. So this could alter the file encoding.

Why did you make this change?

Reply:
Because  ``get_text_with_eol`` converted the strings to Unicode with ``to_text_string`` before the commit https://github.com/spyder-ide/spyder/pull/15903/commits/dde3edc5b460f1ff52bb1671c91236b60f5e0f2b, and this was conflicting with ``toPlainText`` return string, so I removed it from ``get_text_with_eol`` as I needed them to be the same. Hence, I used ``to_text_tring`` in other to keep the same behavior in ``_write_to_file``  after the modifications in ``get_text_with_eol``. If this behavior in ``_write_to_file`` is a newly found bug, I can remove ``to_text_string``, but I don't know if this should be in another PR or Issue.

Reply:
> and this was conflicting with toPlainText return string

In what way was conflicting with toPlainText? I guess toPlainText returns the text in the current encoding and not in unicode?

Reply:
Exactly, to keep the expected encoding in the LSP methods with the change in https://github.com/spyder-ide/spyder/pull/15903/commits/7f820fddf939c98dc14ad0a4669585014aca21d7. It was needed that the ``gest_text_with_eol`` returned the text with the current encoding and not always Unicode.

Reply:
Ok, thanks for the clarification. I think this looks good to me, except for some changes in `get_text_with_eol` I'll mention in an additional review.